src/main/java/datawave/util/time/DateHelper.java

```diff
@@ -213,7 +265,51 @@ public static String format8601(Date date) {
      * @return the {@code Date} object
      */
     public static Date parse8601(String date) {
-        return Date.from(ZonedDateTime.parse(date, DTF_8601).toInstant());
+        try {
+            return Date.from(ZonedDateTime.parse(date, DTF_8601).toInstant());
+        } catch (DateTimeParseException e) {
+            log.warn(e.getMessage());
+            throw e;
+        }
+    }
+    
+    /**
+     * Converts a String in yyyyMMddHHmmss.SSS format to a Date object in a consistent way not dependent on local settings for calendar, timezone, or locale by
+     * using Zulu timezone and US locale.
+     *
+     * @param date
+     * @return the {@code Date} object
+     */
+    public static Date parseRemove(String date) {
+        return lenientParseHelper(date, DTF_Remove, DATE_FORMAT_REMOVE_CONSTANT, true);
+    }
+    
+    /**
+     * Converts a String in simple pattern of letters and symbols described in DateTimeFormatter class documentation to a Date object in a consistent way not
+     * dependent on local settings for calendar, timezone, or locale by using Zulu timezone and US locale.
+     * 
+     * @see DateTimeFormatter
+     *
+     * @param date
+     * @param pattern
+     * @return the {@code Date} object
+     */
+    public static Date parseCustom(String date, String pattern) {
+        final String hourRegex = "(?i)(.*([kh]).*)";
+        
+        // handle a special case where the pattern in yyyyDDD but the day of year is not zero padded
+        // i.e. 202311 should return Jan 11 2023
+        // also assumes date is not a lenient date
+        if ("yyyyDDD".equals(pattern) && "yyyyDDD".length() > date.length()) {
+            pattern = "yyyy D";
+            date = (date.length() > 5) ? new StringBuilder(date).insert(4, " ").toString() : date;
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(pattern).withZone(ZoneOffset.UTC);
+            return Date.from(LocalDate.parse(date, formatter).atStartOfDay(formatter.getZone()).toInstant());
+        }
+        
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(pattern).withZone(ZoneOffset.UTC);
+        
+        return lenientParseHelper(date, formatter, pattern, pattern.matches(hourRegex));
```

Comment:
Not sure how often we call this but if it's a non-trivial amount of times (like once per event during ingest/query), we could get a decent savings by making a constant for the compiled `hourRegex`. I believe the `matches()` call will have to recompile it every time.

Reply:
Done.